app/serializers/book_serializer.rb

```diff
@@ -1,7 +1,10 @@
-class BookSerializer < BaseSerializer
-  attributes :name,:author , :release_date ,:category_count
-  
-  attribute :category_count do |object|
-    object.with_category_count
+class BookSerializer < BaseSerializer      
+  attributes :name , :author ,:release_date
+
+  def name
```

Comment:
its better to define an attribute and pass your object as a block incase you need to handle stuff based on passed params rather than defining sit in stones functions